### PR TITLE
feat(provider): inject bashrc into instance

### PR DESCRIPTION
**Pull request by syu-w**
This allowed to save some info and environment variable to help debug when needed.

### CODE REVIEW

File: craft_application/services/provider.py [hunk 1]
```diff
@@ -233,3 +237,20 @@ def _capture_logs_from_instance(self, instance: craft_providers.Executor) -> Non
...
+
+    def _setup_instance_bashrc(self, instance: craft_providers.Executor) -> None:
+        """Set up the instance's bashrc to export environment."""
+        bashrc = pkgutil.get_data("craft_application", "misc/instance_bashrc")
+
+        if bashrc is None:
+            emit.debug(
+                "Could not find the bashrc file in the craft-application package"
+            )
+            return
```
Comment (syu-w) [inline craft_application/services/provider.py:249]: I am not sure if we should error out or just a warning here

Comment (mr-cal) [reply craft_application/services/provider.py:249]: I'm not sure either, I'm OK with either.

**Review by mr-cal — APPROVED** [review]
:+1:

**Review by lengau — APPROVED** [review]
Thanks!

File: craft_application/services/provider.py (hunk 1 above)
Comment (syu-w) [reply craft_application/services/provider.py:249]: Added the test.

**Pull request merged**